rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java

```diff
@@ -1534,6 +1534,34 @@ void should_UpdateTaskOwnerOfReadyTask() {
     assertThat(theUpdatedTaskRepresentationModel.getOwner()).isEqualTo("dummyUser");
   }
 
+  @Test
+  void should_UpdateTaskOwnerOfReadyForReviewTask() {
+    final String url = restHelper.toUrl("/api/v1/tasks/TKI:000000000000000000000000000000000104");
+    HttpEntity<Object> auth = new HttpEntity<>(RestHelper.generateHeadersForUser("user-1-2"));
+
+    // retrieve task from Rest Api
+    ResponseEntity<TaskRepresentationModel> responseGet =
+        TEMPLATE.exchange(url, HttpMethod.GET, auth, TASK_MODEL_TYPE);
+
+    assertThat(responseGet.getBody()).isNotNull();
+    TaskRepresentationModel theTaskRepresentationModel = responseGet.getBody();
+    assertThat(theTaskRepresentationModel.getState()).isEqualTo(TaskState.READY_FOR_REVIEW);
+    assertThat(theTaskRepresentationModel.getOwner()).isNull();
+
+    // set Owner and update Task
+    theTaskRepresentationModel.setOwner("dummyUser");
+    HttpEntity<TaskRepresentationModel> auth2 =
+        new HttpEntity<>(theTaskRepresentationModel, RestHelper.generateHeadersForUser("user-1-2"));
+
+    ResponseEntity<TaskRepresentationModel> responseUpdate =
+        TEMPLATE.exchange(url, HttpMethod.PUT, auth2, TASK_MODEL_TYPE);
+
+    assertThat(responseUpdate.getBody()).isNotNull();
+    TaskRepresentationModel theUpdatedTaskRepresentationModel = responseUpdate.getBody();
+    assertThat(theUpdatedTaskRepresentationModel.getState()).isEqualTo(TaskState.READY_FOR_REVIEW);
+    assertThat(theUpdatedTaskRepresentationModel.getOwner()).isEqualTo("dummyUser");
+  }
```

Comment:
Why are you testing this on REST level? I expected a test in taskana-core or taskana-core-test

Reply:
I'll add a test there, but would leave this test as we are also testing it for state READY on REST level.